src/Workspaces/Core/Portable/Workspace/Host/Mef/ExportWorkspaceServiceAttribute.cs

```diff
@@ -31,16 +31,21 @@ public class ExportWorkspaceServiceAttribute(Type serviceType, string layer = Se
         public string Layer { get; } = layer ?? throw new ArgumentNullException(nameof(layer));
 
         /// <summary>
-        /// <see cref="WorkspaceKind"/>s that the service is specified for.
+        /// <see cref="WorkspaceKinds"/> that the service is specified for.
         /// If non-empty the service is only exported for the listed workspace kinds and <see cref="Layer"/> is not applied,
         /// unless <see cref="Layer"/> is <see cref="ServiceLayer.Test"/> in which case the export overrides all other exports.
         /// </summary>
-        internal IReadOnlyList<string> WorkspaceKinds { get; } = [];
+        internal WorkspaceKinds WorkspaceKinds { get; } = WorkspaceKinds.Unknown;
```

Comment:
Why not just convert the enum to a list of the string and then leave the rest of the code untouched? I think then it'd mean the metadata gets cached directly that way.

Reply:
We could, but this seems cleaner. The complexity of both is similar.